src/main/java/org/opensearch/plugin/insights/settings/QueryInsightsSettings.java

```diff
@@ -69,6 +70,11 @@ public class QueryInsightsSettings {
      */
     public static final String PLUGINS_BASE_URI = "/_insights";
 
+    public static final GroupingType DEFAULT_GROUPING_TYPE = GroupingType.NONE;
+    public static final int DEFAULT_MAX_GROUPS = 100;
+
+    public static final int MAX_GROUPS_LIMIT = 10000;
```

Comment:
Can we define the GROUPS limit as function of the number of queries being tracked by top N? 5-10x that limit should be pretty reasonable IMO. Also, can we rename `DEFAULT_MAX_GROUPS` to `DEFAULT_GROUPS_LIMIT`?

Reply:
We currently manage this as a cluster setting that can be adjusted manually, offering flexibility to increase or decrease the limit as needed. Relying solely on a function of TopN may not be ideal. While setting Max N at 100 would result in a maximum of 1000 groups using the 10x multiplier, this may be insufficient for domains with a high volume of unique queries within the given window. Keeping it adjustable allows for better adaptability to different workloads. 

Sure will do the rename.